python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -946,12 +958,32 @@ def _make_request(
         raise Failure(f"Max retries ({self.request_max_retries}) exceeded with url: {url}.")
 
     def get_connections(self) -> Mapping[str, Any]:
-        """Fetches all connections of an Airbyte workspace from the Airbyte API."""
-        raise NotImplementedError()
+        """Fetches all connections of an Airbyte workspace from the Airbyte REST API."""
+        return self._make_request(
+            method="GET",
+            endpoint="connections",
+            base_url=self.rest_api_base_url,
+            params={"workspaceIds": self.workspace_id},
+        )
+
+    def get_connection_details(self, connection_id) -> Mapping[str, Any]:
+        """Fetches details about a given connection from the Airbyte Configuration API."""
+        # Using the Airbyte Configuration API to get the connection details, including streams and their configs.
```

Comment:
We use the Airbyte Configuration API only when required. According to their docs, this is an internal API that can be changed without notice, see [here](https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#overview).

We have been using it for the legacy Airbyte OSS resource. It is required here as well because the connections/{connection_id} endpoint in the public API does not return the stream configs in the connection details, which is required to know whether a stream is selected (enabled).

This seems like fairly low risk since we have been using the endpoint in the Configuration API for the past 2 years.

Reply:
let's specifically note that the api is internal and could potentiall.y change.